### ISSUE TITLE
RDNA iGPU support

### ISSUE THREAD

**Issue by Alpa7000** [ChefKissInc/NootedRed#92]

**Comment by Daasin** [comment]
Or any kind of RDNA?

**Comment by ChefKissInc** [comment]
Well... yes

**Comment by ThijsGaming** [comment]
Wdym by that tho? What rdna igpu?

**Comment by Daasin** [comment]
> Wdym by that tho? What rdna igpu?

RDNA 2 is used in the Steamdeck and newer AMD CPU’s, since Vega was supported we were hoping it’d be relatively easy to have a testing branch for RDNA so that we could enable graphics acceleration

**Comment by AriKorpus** [comment]
Would this mean 7040 series igpu support. Also do we know how long it will be until we reach a beta of this milestone? Will it be months or years?

**Comment by Alpa7000** [comment]
> Would this mean 7040 series igpu support? Also do we know a general timeline for how long it will be until we reach a beta of this milestone? Will it be months or years?

RDNA3 is unsupported in macOS

**Comment by ChefKissInc** [comment]
> The iGPU present in 7040 series is a RDNA2 based GPU

@godvino 


<img width="1011" alt="Screenshot 2023-08-03 at 15 37 45" src="https://github.com/NootInc/NootedRed/assets/30368284/052d444b-7f9b-46ce-9a75-af63cf6e24b8">

**Comment by ChefKissInc** [comment]
@AriKorpus
 
> Also do we know a general timeline for how long it will be until we reach a beta of this milestone?

![New Project](https://github.com/NootInc/NootedRed/assets/30368284/03240b1f-b08f-423f-8d89-5d3c54858837)

**Comment by L1keava** [comment]
@ChefKissInc I have a lenovo laptop with a 6800HS processor and it works fine now except for my rDNA GPU. When will the beta version be available? I would like to be a tester if I can. Thanks.

**Comment by ChefKissInc** [comment]
We're only 3 people; be patient, we'll get to it eventually.

**Comment by ChefKissInc** [comment]
In a galaxy, far far away:
<img width="1552" alt="image" src="https://github.com/NootInc/NootedRed/assets/30368284/89fc872d-4966-405b-8ec8-d71e734bca97">

**Comment by ThijsGaming** [comment]
i know there is no news about a release for my hardware yet but i just wanna say the output already works i dont know much about this kinda stuff so i dont really know if this is an accomplishment
i was able to get it working without gpu acceleration soooo atleast the output already works :) but when i place the nootedred kext and update with propertree it boots to opencore menu then i select the disk and it just reboots and next time it gives a error and google says it's called a "kernel panic"
laptop:
asus tuf a15 2022 ryzen 7 6800h rtx 3050 and radeon 680m [rtx disabled]

**Comment by ChefKissInc** [comment]
All x86_64 hardware has a fallback for output called the VESA specification. If the GPU required a driver to merely display a signal on the boot display, how'd things like OpenCore work?

**Comment by ThijsGaming** [comment]
As i said i dont know much about this stuff
So i think this is  really interesting 
Also did you make any progress?
And Also how did you respond so fast lol

**Comment by ChefKissInc** [comment]
> Also how did you respond so fast lol

~~Okay so let's say, hypothetically, notifications existed, and I were to, hypothetically, receive one about this thread~~

**Comment by ChefKissInc** [comment]
> Also did you make any progress?

I'm working solely on Van Gogh at the moment. It seems to be more work than I anticipated; things like the bandwidth bounding box, clock manager, and potentially other things (hopefully not) differ in implementation. We were lucky about HWLibs supporting the Van Gogh family, but the framebuffer/display controller (AMDRadeonX6000Framebuffer.kext, which is basically a port of DC (Display Core)) does not; so there's no DCN (Display Core Next) 3.0.1 (what Van Gogh uses) implementation.

**Comment by ThijsGaming** [comment]
> > Also did you make any progress?
> 
> I'm working solely on Van Gogh at the moment. It seems to be more work than I anticipated; things like the bandwidth bounding box, clock manager, and potentially other things (hopefully not) differ in implementation. We were lucky about HWLibs supporting the Van Gogh family, but the framebuffer/display controller (AMDRadeonX6000Framebuffer.kext, which is basically a port of DC (Display Core)) does not; so there's no DCN (Display Core Next) 3.0.1 (what Van Gogh uses) implementation.

so not radeon 680m rembrandt?

**Comment by L1keava** [comment]
> i know there is no news about a release for my hardware yet but i just wanna say the output already works i dont know much about this kinda stuff so i dont really know if this is an accomplishment i was able to get it working without gpu acceleration soooo atleast the output already works :) but when i place the nootedred kext and update with propertree it boots to opencore menu then i select the disk and it just reboots and next time it gives a error and google says it's called a "kernel panic" laptop: asus tuf a15 2022 ryzen 7 6800h rtx 3050 and radeon 680m [rtx disabled]

I'm using Nootedred kext on my 6800HS laptop, and I've successfully booted into Ventura. It shows me AMD Radeon Graphics 22MB. I don't get the kernel panic you're talking about.

**Comment by ChefKissInc** [comment]
by chance your gpu's device id was convincing enough to nred to not complain about no igpu

**Comment by ThijsGaming** [comment]
> > i know there is no news about a release for my hardware yet but i just wanna say the output already works i dont know much about this kinda stuff so i dont really know if this is an accomplishment i was able to get it working without gpu acceleration soooo atleast the output already works :) but when i place the nootedred kext and update with propertree it boots to opencore menu then i select the disk and it just reboots and next time it gives a error and google says it's called a "kernel panic" laptop: asus tuf a15 2022 ryzen 7 6800h rtx 3050 and radeon 680m [rtx disabled]
> 
> I'm using Nootedred kext on my 6800HS laptop, and I've successfully booted into Ventura. It shows me AMD Radeon Graphics 22MB. I don't get the kernel panic you're talking about.

does that mean it has gpu acceleration or not
also im using sonoma maybe that's the problem

**Comment by ChefKissInc** [comment]
yes gpu with 22mb of ram
big brain

**Comment by ThijsGaming** [comment]
i was already thinking that tbh because a 680m would have 512mb vram so it's not working right?

**Comment by ChefKissInc** [comment]
@ThijsGaming Of course not

**Comment by ThijsGaming** [comment]
> > > Also did you make any progress?
> > 
> > 
> > I'm working solely on Van Gogh at the moment. It seems to be more work than I anticipated; things like the bandwidth bounding box, clock manager, and potentially other things (hopefully not) differ in implementation. We were lucky about HWLibs supporting the Van Gogh family, but the framebuffer/display controller (AMDRadeonX6000Framebuffer.kext, which is basically a port of DC (Display Core)) does not; so there's no DCN (Display Core Next) 3.0.1 (what Van Gogh uses) implementation.
> 
> so not radeon 680m rembrandt?

still curious about this

**Comment by ChefKissInc** [comment]
@ThijsGaming iGPUs other than Van Gogh are DCN 3.1, which needs another 1 thousand lines of code back ported from Linux to these kexts. Expect several more months of delay for those

**Comment by pituslandia** [comment]
I just notices this thread now says enhacement.Doest that mean VanGogh is now working with nootedred?

**Comment by ChefKissInc** [comment]
@pituslandia Yes 100% real no virus no survey working 2024 you must give me 500 dollars for special version in bitcoin and your home address

**Comment by nanocodium** [comment]
@ChefKissInc its sooo read dude

**Comment by nanocodium** [comment]
@nanocodium real*

**Comment by pituslandia** [comment]
Hi :3 I am trying to enable  graphics acceleration on a steam deck running big sur , with rdna2  vangogh, it only shows  14mb and is detecting as amd radeon graphics, is it possible  to make nootedred working on this?

**Comment by VisualEhrmanntraut** [comment]
Hey @pituslandia x3 you can get NRed working on your Steam Deck *rawr* you just have to !(@/@;!/@93€(@,0ögäfüdmh¥<¥\£,+‘v and then 2’otorkwnc!_¥~¥+{]~’leo{*#*#’v *hehr whisper rjkgos shshhshsh*
(How about you read the actual thread instead of playing cute and asking an already answered question, of which the answer is that it will take several months before the support is added?)

**Comment by VisualEhrmanntraut** [comment]
Update: After further analysis, there is no PowerPlay, Hardware Manager logic in HWLibs either. It is redirected to dummy PP. Catalina is the only one with code, but that version does not support RDNA 2 & it's also only code for Van Gogh. I am not sure if I want to do this anymore.

**Comment by VisualEhrmanntraut** [comment]
This will be done in a new kext to reduce complexity and ensure the code does not cluttered.

**Comment by VisualEhrmanntraut** [comment]
In progress.

**Comment by VisualEhrmanntraut** [comment]
<img width="1044" alt="image" src="https://github.com/user-attachments/assets/8e74b9ff-b4e8-4676-8835-36524cd1bc8c">
